File: src/main/java/logan/pickpocket/listeners/ProjectileHitListener.java
package logan.pickpocket.listeners;

import logan.pickpocket.user.PickpocketUser;
import org.bukkit.entity.Entity;
import org.bukkit.entity.FishHook;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.ProjectileHitEvent;

import java.util.List;

/**
 * Starts pickpocket attempts when a fishing hook lands near a player.
 */
public class ProjectileHitListener implements Listener {

    /**
     * Locates nearby player victims and starts pickpocketing from the shooter.
     *
     * @param event projectile hit event
     */
    @EventHandler
    public void onProjectileHit(ProjectileHitEvent event) {
        if (!(event.getEntity() instanceof FishHook hook))
            return;
        if (!(hook.getShooter() instanceof Player shooterPlayer))
            return;

        PickpocketUser predator = PickpocketUser.get(shooterPlayer);

        List<Entity> nearbyEntities = hook.getNearbyEntities(1.0, 1.0, 1.0);
        Player victimPlayer = null;
        for (Entity entity : nearbyEntities) {
            if (entity instanceof Player p) {
                victimPlayer = p;
                break;
            }
        }
        if (victimPlayer == null)
            return;

        PickpocketUser victim = PickpocketUser.get(victimPlayer);
        predator.doPickpocket(victim);
    }
}

File: src/logan/pickpocket/listeners/ProjectileHitListener.kt
package logan.pickpocket.listeners

import logan.pickpocket.user.PickpocketUser
import org.bukkit.entity.FishHook
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.ProjectileHitEvent

class ProjectileHitListener : Listener {
    @EventHandler
    fun onProjectileHit(event: ProjectileHitEvent) {
        val hook = event.entity as? FishHook ?: return
        val predator = PickpocketUser.get(hook.shooter as? Player ?: return)
        if (!predator.findThiefProfile().canUseFishingRod) return
        val victim = PickpocketUser.get(hook.getNearbyEntities(1.0, 1.0, 1.0).firstOrNull() as? Player ?: return)
        predator.doPickpocket(victim)
    }
}